operator_manifest/utils.py

```diff
@@ -0,0 +1,145 @@
+        digest = skopeo_raw_json['manifests'][0]['digest']
+        return f'{name}@{digest}'
+
+    if (
+        skopeo_raw_json.get('mediaType') == 'application/vnd.docker.distribution.manifest.v2+json'
+        and skopeo_raw_json.get('schemaVersion') == 2
+    ):
+        raw_digest = hashlib.sha256(skopeo_raw.encode('utf-8')).hexdigest()
+        return f'{name}@sha256:{raw_digest}'
+
+    error_msg = (
+        f'The pull specification of {resolve_bundle_digest} is neither a v2 manifest list nor a'
+        f'v2 schema 2 image manifest. Type {skopeo_raw_json.get("mediaType")} and schema version '
+        f'{skopeo_raw_json.get("schemaVersion")} is not supported.'
+    )
+    raise ValueError(error_msg)
+
+
+def _get_container_image_name(pull_spec):
+    """
+    Get the container image name from a pull specification.
+
+    :param str pull_spec: the pull spec to analyze
+    :return: the container image name
+    """
+    if '@' in pull_spec:
+        return pull_spec.split('@', 1)[0]
+    else:
+        return pull_spec.rsplit(':', 1)[0]
+
+
+# TODO: Make this nicer and more relevant
+def retry(attempts=3, wait_on=Exception):
+    """
+    Retry a section of code until success or max attempts are reached.
+
+    :param int attempts: the total number of attempts to make before erroring out
+    :param Exception wait_on: the exception on encountering which the function will be retried
+    :raises Exception: if the maximum attempts are reached
+    """
+
+    def wrapper(function):
+        @functools.wraps(function)
+        def inner(*args, **kwargs):
+            remaining_attempts = attempts
+            while True:
+                try:
+                    return function(*args, **kwargs)
+                except wait_on:
+                    remaining_attempts -= 1
+                    if remaining_attempts <= 0:
+                        raise
+
+        return inner
+
+    return wrapper
+
+
+@retry(wait_on=ValueError)
+def skopeo_inspect(*args, return_json=True, require_media_type=False):
+    """
+    Wrap the ``skopeo inspect`` command.
+
+    :param args: any arguments to pass to ``skopeo inspect``
+    :param bool return_json: if ``True``, the output will be parsed as JSON and returned
+    :param bool require_media_type: if ``True``, ``mediaType`` will be checked in the output
+        and it will be ignored when ``return_json`` is ``False``
+    :return: a dictionary of the JSON output from the skopeo inspect command
+    :rtype: dict
+    :raises ValueError: if the command fails and if ``mediaType`` is not found in the output while
+        ``require_media_type`` is ``True``
+    """
+    exc_msg = None
+    for arg in args:
+        if arg.startswith('docker://'):
+            exc_msg = f'Failed to inspect {arg}. Make sure it exists and is accessible.'
+            break
+
+    # TODO: Make this configurable?
+    skopeo_timeout = '300s'
+    cmd = ['skopeo', '--command-timeout', skopeo_timeout, 'inspect'] + list(args)
```

Comment:
maybe just `--config` to avoid fetching all tags. This may not be needed as we may backport also code from OSBS to handle registry queries, not designed yet

Reply:
It should be up to the caller. The one call to this method uses `--raw` which also ignores the tags.